mesh/generator/generator.go

```diff
@@ -185,6 +185,65 @@ func BuildMeshMap(ctx context.Context, o mesh.Options, gi *mesh.GlobalInfo) (mes
 			mesh.CheckError(err)
 		}
 
+		// if ambient, add ztunnel
+		if gi.KialiCache.IsAmbientEnabled(cp.Cluster.Name) {
+			ztunnels, err := gi.Business.Workload.GetAllWorkloads(ctx, cp.Cluster.Name, fmt.Sprintf("%s=%s", config.IstioAppLabel, config.Ztunnel))
```

Comment:
I'm not sure if we require fresh data here but if not, I think that could be used instead? 
`ztunnelPods := gi.KialiCache.GetZtunnelPods(cp.Cluster.Name)`

Reply:
You are right that this is probably not the most efficient call here, because we are only interested in daemonsets for ztunnel. But looking at the code, under the covers this call does use the cache (except if the user has enabled some special workload types).  I could use the dedicated ztunnel call you mention, but I think I would have to add some logic to get the workload. Also, I'm not sure, but does the direct cache call consider the user's accessible namespaces?  I think the call as-is should be fine, I don't think there will be a perf issue here. So, I will leave it for now.  If we create a specific call to get ztunnel workloads, then we can change it later.